core/model/src/test/java/org/eclipse/sw360/antenna/model/license/WithLicenseTest.java

```diff
@@ -88,6 +88,33 @@ public void testWithLicenseBuildFromAssembledInput() {
         assertThat(withLicense1.equalLicense(withLicense2)).isTrue();
     }
 
+    @Test(expected=IllegalArgumentException.class)
```

Comment:
A general remark: According to the commit message, only this file should be changes, and all changes in other files should be separated out to other commit with proper commit messages.

Reply:
See my answer above, the other stuff were code smells reported by SonarQube which I fixed as the commit message says. You could argue, that both things could be separated into two commits, but I do not see here the advantage of having this separated, because the changes are not really changing a lot.

Reply:
The advantage of separating semantically different changes into different commits for me as a reviewer is that I can skip over commits that say to change things which are either unimportant to me, or that are things I cannot comment on.

So, if a commit says to just add a test, that hardly breaks the main app, and I might skip the review if I'm short of time. Similarly, if a commit says to just add comments, that is a non-functional change which I can just skim over and proceed to the next commit.

However, for that to work I need to be able to rely on the commit message to really only change what is stated here. At least when the commit comes from a trusted contributor. For untrusted parties I of course review everything, thoroughly :-)